cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/IRVariable.qll

```diff
@@ -55,37 +54,42 @@ abstract class IRVariable extends TIRVariable {
   /**
    * Gets the function that references this variable.
    */
-  final Function getEnclosingFunction() {
+  final Language::Function getEnclosingFunction() {
     result = func
   }
 }
 
 /**
  * Represents a user-declared variable referenced by the IR for a function.
  */
-abstract class IRUserVariable extends IRVariable {
-  Variable var;
+class IRUserVariable extends IRVariable, TIRUserVariable {
+  Language::Variable var;
+  Language::Type type;
 
-  override final string toString() {
-    result = var.toString()
+  IRUserVariable() {
+    this = TIRUserVariable(var, type, func)
   }
 
-  override final Type getType() {
-    result = getVariableType(var)
+  override final string toString() {
+    result = getVariable().toString()
   }
 
-  override final Locatable getAST() {
+  override final Language::AST getAST() {
     result = var
   }
 
   override final string getUniqueId() {
-    result = var.toString() + " " + var.getLocation().toString()
+    result = getVariable().toString() + " " + getVariable().getLocation().toString()
+  }
+
+  override final Language::Type getType() {
+    result = type
   }
 
   /**
    * Gets the original user-declared variable.
    */
-  final Variable getVariable() {
+  Language::Variable getVariable() {
```

Comment:
This lost its `final`. I assume that wasn't intentional?

Reply:
The subclasses override it just to provide a more specific result type.